src/web/services/search.service.ts

```diff
@@ -342,16 +342,25 @@ export class SearchService {
     };
     for (const feedbackSession of feedbackSessions.feedbackSessions) {
       if (this.feedbackSessionService.isFeedbackSessionOpen(feedbackSession)) {
-        feedbackSessionLinks.openSessions[this.feedbackSessionService.generateNameFragment(feedbackSession).toString()]
-          = this.linkService.generateSubmitUrl(student, feedbackSession.feedbackSessionName);
+        feedbackSessionLinks.openSessions[feedbackSession.feedbackSessionName]
+            = {
+              ...this.feedbackSessionService.formatProperties(feedbackSession),
+              feedbackSessionUrl: this.linkService.generateSubmitUrl(student, feedbackSession.feedbackSessionName),
```

Comment:
I would even go this far and say that `LinkService` should be deprecated as this class is the only user, but you don't need to change this now.